### PR TITLE
add active_experiments map to TargetingAttributes and verify jexl interactions

### PR DESCRIPTION
**Pull request by jeddai**
- Update jexl-eval to 0.2.2
- Add active_experiments  map to TargetingAttributes
- Verify jexl interactions with the active_experiments map

[EXP-2545](https://mozilla-hub.atlassian.net/browse/EXP-2545)

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ac: android-components-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

File:   [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (tarikeshaq) [inline  :1]: hmmm not sure what this is, but seems like an empty file with a new line?

File: components/nimbus/src/tests/test_evaluator.rs [hunk 2]
```diff
@@ -423,6 +423,98 @@ fn test_targeting_is_already_enrolled() {
     );
 }
 
+#[test]
+fn test_targeting_active_experiments_equivalency() {
```
Comment (tarikeshaq) [inline components/nimbus/src/tests/test_evaluator.rs:427]: Thanks for adding the tests!! it would be great to test it end-to-end when we have the active experiments loaded into the targeting attributes but that could be a separate PR

(i.e we create a nimbus client, enroll it in an experiment, then verify that targeting by `active_experiments` behaves as expected)

File: CHANGES_UNRELEASED.md [hunk 3]
```diff
@@ -21,4 +21,7 @@ Use the template below to make assigning a version number during the release cut
 ## Nimbus FML
 
 ### What's Changed
-  - Add `channels` value for defaults and add support for multiple channels in `channel` via comma separation. ([#5101](https://github.com/mozilla/application-services/pull/5101))
\ No newline at end of file
+  - Add `channels` value for defaults and add support for multiple channels in `channel` via comma separation. ([#5101](https://github.com/mozilla/application-services/pull/5101))
+
+### What's New
+  - JEXL targeting allows for using the `in` keyword with objects and a map of active experiments has been added to TargetingAttributes ([#5104](https://github.com/mozilla/application-services/pull/5104))
```
Comment (tarikeshaq) [inline CHANGES_UNRELEASED.md:27]: Maybe a note that the map is currently always empty?

**Review by tarikeshaq — CHANGES_REQUESTED** [review]
Awesome @jeddai 🚀 !! Requesting changes for the empty file and a couple of questions I had about loading the active experiments into the `TargetingAttributes` - otherwise looks great!!

File:   (hunk 1 above)
Comment (jeddai) [reply  :1]: Same. My diff for this commit shows it as `/dev/null` before, and `b/` after. I may need to cherrypick changes

Comment (jeddai) [reply  ]: Fixed!

File: CHANGES_UNRELEASED.md (hunk 3 above)
Comment (jeddai) [reply CHANGES_UNRELEASED.md]: Done!

**Review by tarikeshaq — CHANGES_REQUESTED** [review]
Awesome @jeddai!!!

the swiftlint job is a mac runner problem, I have a fix in https://github.com/mozilla/application-services/pull/5113 you can ignore it it'll land eventually and your change doesn't change swift.

As for the dependency check job it popped up because I imagine you ran `cargo update`?

We have two options:
1. Apply the fixup for the broken dependency summary (could be a good exercise to go and find out how we report dependency summaries!)
1. Undo the `cargo update` (i.e revert the changes to `Cargo.lock` and only bump the `jexl` dependency. Then in a separate PR we can run the upgrade and fixup the dependency summaries.

Also, can you file a followup ticket for actually loading the experiments into the `TargetingAttributes`?

**Comment by jeddai** [comment]
I reverted the `cargo update` and it should be good now!

**Comment by jeddai** [comment]
JIRA ticket for setting up TargetingAttributes https://mozilla-hub.atlassian.net/browse/EXP-2735

**Review by tarikeshaq — APPROVED** [review]
Awesome!!!


Land away! 🚢 🚢
